spyder_line_profiler/spyder/widgets.py

```diff
@@ -100,123 +100,214 @@ def natural_sort_key(key):
+        self.error_output = None
+        self.output = None
+        self.use_colors = True
+        self.spyder_pythonpath = None
+        self.process = None
+        self.started_time = None
+        
+        # Widgets
+        self.filecombo = PythonModulesComboBox(
+            self, id_= SpyderLineProfilerWidgetMainToolbarItems.FileCombo)
+        self.datatree = LineProfilerDataTree(self)
+        self.datelabel = QLabel(self)
+        self.datelabel.ID = SpyderLineProfilerWidgetInformationToolbarItems.DateLabel
+        self.datelabel.setText(_('Please select a file to profile, with '
+                                 'added @profile decorators for functions'))
+        self.timer = QTimer(self)
 
-        self.filecombo = PythonModulesComboBox(self)
-
-        self.start_button = create_toolbutton(
-            self, icon=get_icon('run.png'),
+        layout = QVBoxLayout()
+        layout.addWidget(self.datatree)
+        self.setLayout(layout)
+        
+        # Signals
+        self.datatree.sig_edit_goto_requested.connect(
+            self.sig_edit_goto_requested)
+        
+    # --- PluginMainWidget API
+    # ------------------------------------------------------------------------
+    def get_title(self):
+        return _("Line Profiler")
+
+    def get_focus_widget(self):
+        pass
+
+    def setup(self):
+        
+        self.start_action = self.create_action(
+            SpyderLineProfilerWidgetActions.Run,
             text=_("Profile by line"),
             tip=_("Run line profiler"),
-            triggered=(lambda checked=False: self.analyze()), text_beside_icon=True)
-        self.stop_button = create_toolbutton(
-            self,
-            icon=get_icon('terminate.png'),
+            icon=self.create_icon('run'),
+            triggered=self.start,
+        )
+        self.stop_action = self.create_action(
+            SpyderLineProfilerWidgetActions.Stop,
             text=_("Stop"),
             tip=_("Stop current profiling"),
-            text_beside_icon=True)
-        self.filecombo.valid.connect(self.start_button.setEnabled)
-        #self.filecombo.valid.connect(self.show_data)
-        # FIXME: The combobox emits this signal on almost any event
-        #        triggering show_data() too early, too often.
-
-        browse_button = create_toolbutton(
-            self, icon=get_icon('fileopen.png'),
-            tip=_('Select Python script'),
-            triggered=self.select_file)
-
-        self.datelabel = QLabel()
-
-        self.log_button = create_toolbutton(
-            self, icon=get_icon('log.png'),
-            text=_("Output"),
-            text_beside_icon=True,
+            icon=self.create_icon('stop'),
+            triggered=self.kill_if_running,
+        )
+        self.browse_action = self.create_action(
+            SpyderLineProfilerWidgetActions.Browse,
+            text=_("Open Script"),
+            tip=_('Select Python script'),            
```

Comment:
Remove trailing whitespace:
```suggestion
            tip=_('Select Python script'),
```